app.js

```diff
@@ -26,10 +66,17 @@ app.use(function(req, res, next) {
 });
 
 app.get('/validate', function (req, res, next) {
-  const token = firebase.auth().createCustomToken(req.query.token);
-  res.json({ token });
+  var apiToken = req.query.token;
+  if (isValidToken(apiToken)) {
+    getApiUser(apiToken).then(function() {
+      const token = firebase.auth().createCustomToken(req.query.uid);
```

Comment:
and yeah if you have the api user then you can use the user.login string as the UID here.